gamemodes/terrortown/gamemode/shared/events/base_event.lua

```diff
@@ -0,0 +1,35 @@
+EVENT.type = "base_event"
+EVENT.event = {}
+
+function EVENT:AddData(event)
+	self.event = event
+end
+
+function EVENT:Add(event)
+	-- store the event time in relation to the round start time in milliseconds
+	event.time = math.Round((CurTime() - GAMEMODE.RoundStartTime) * 1000, 0)
+	event.roundState = GetRoundState()
+
+	-- call hook that a new event is about to be added, can be canceled or
+	-- modified from that hook
+	if hook.Run("TTT2OnTriggeredEvent", self.type, event) == false then
+		return false
+	end
+
+	self:AddData(event)
+
+	-- after the event is added, it should be passed on to the
+	-- scoring function to directly calculate the score
+	if isfunction(self.Score) then
+		self:Score(self.event)
```

Comment:
This is something that we have to discuss. I used it as a parameter because it makes it easier (in my opinion) for people using those events. However I'm open to switching to a `self.events` reference here.